src/response.rs

```diff
@@ -0,0 +1,59 @@
+use core::fmt::{Debug, Write};
+use serde::{Deserialize, Serialize};
+
+use heapless::{String, Vec};
+
+// The maximum size of a response.
+const MAX_RESPONSE_SIZE: usize = 128;
+
+/// Responses are always generated as a result of handling an in-bound request.
+#[derive(Debug, Default, Clone, Deserialize, Serialize)]
+pub struct Response {
```

Comment:
While I think it could be valuable to make this generic, it also adds a lot of code and type complexity on the end user by doing so.

If this were generic, that would add another generic type to the `Minireq` type, which already has a ton of generics. I also worry this may bubble up to the end user and require them to specify `Response::<SIZE>::ok()` when generating return types from the handlers, but I haven't tested it. I'll address all the other review points and then try making this generic to see how it works out.

Reply:
After testing, I don't see why we can't make it generic and then link it directly to the `MESSAGE_SIZE` generic, since that is an upper-bound on the maximum response size anyways. This gets around having any type bloat.